heartbeat/_meta/fields.yml

```diff
@@ -2,131 +2,257 @@
-          description: Duration in microseconds
+        - name: rtt
+          type: group
+          description: >
+            TLS layer round trip times.
+          fields:
+            - name: connect
+              type: group
+              description: >
+                Time required to establish a connection via SOCKS5 to endpoint
+                based on available connection to SOCKS5 proxy.
+              fields:
+                - name: us
+                  type: long
+                  description: Duration in microseconds
+
 
-    - name: validate_rtt
+
+
+    - name: tls
       type: group
       description: >
-        Time required for validating the connection if connection checks are configured.
+        TLS layer related fields.
       fields:
-        - name: us
-          type: long
-          description: Duration in microseconds
+        - name: rtt
+          type: group
+          description: >
+            TLS layer round trip times.
+          fields:
+            - name: handshake
+              type: group
+              description: >
+                Time required to finish TLS handshake based on already available network
+                connection.
+              fields:
+                - name: us
+                  type: long
+                  description: Duration in microseconds
 
-    - name: response
+    - name: http
       type: group
       description: >
-        Service response parameters.
-
+        HTTP related fields.
       fields:
-        - name: status
-          type: integer
+        - name: url
+          type: text
           description: >
-            Response status code.
+            Service url used by monitor.
 
-    - name: up
-      required: true
-      type: boolean
-      description: >
-        Boolean indicator if monitor could validate the service to be available.
+        - name: response
+          type: group
+          description: >
+            Service response parameters.
+          fields:
+            - name: status
+              type: integer
+              description: >
+                Response status code.
+        - name: rtt
+          type: group
+          description: >
+            HTTP layer round trip times.
+          fields:
+            - name: validate
+              type: group
+              description: |
+                Duration between first byte of HTTP request being written and
+                response being processed by validator. Duration based on already
+                available network connection.
+                Note: if validator is not reading body or only a prefix, this
+                      number does not fully represent the total time needed
+                      to read the body.
+              fields:
+                - name: us
+                  type: long
+                  description: Duration in microseconds
+
+            - name: validate_body
+              type: group
+              description: |
```

Comment:
oh, oversight from my side. in YAML one can use `>`, `>-` or `|`.

`>`: folded-style. Replace newlines with space. Multiple blank lines are replaced with newline (or space in some implementations)
`|`: literal block. Keeps newlines and blank lines.

I wanted to enforce the newline (or better blank line?) before "Note:" in the hopes of the asciidoc putting the note in a separate new line. That's why I used `|`. 

If I remember correctly `>-` replaces a blank line (`\n\n`) by one newline (`\n`).